archive/2025-09-14-vaultgemma-the-world-s-most-capable-differentially-private-llm.md

```diff
@@ -0,0 +1,200 @@
+- [Send via Email](mailto:name@example.com?subject=Check%20out%20this%20site&body=Check%20out%20https%3A//research.google/blog/vaultgemma-the-worlds-most-capable-differentially-private-llm/ "Send via Email")
+- Copy link
+
+
+
+
+×
+
+
+
+As AI becomes more integrated into our lives, building it with privacy at its core is a critical frontier for the field. [Differential privacy](https://en.wikipedia.org/wiki/Differential_privacy) (DP) offers a mathematically robust solution by adding calibrated noise to prevent memorization. However, applying DP to LLMs introduces trade-offs. Understanding these trade-offs is crucial. Applying DP noise alters traditional [scaling laws](https://arxiv.org/abs/2203.15556) — rules describing performance dynamics — by reducing training stability (the model's ability to learn consistently without experiencing catastrophic events like loss spikes or divergence) and significantly increasing batch size (a collection of input prompts sent to the model simultaneously for processing) and computation costs.
+
+Our new research, “ [Scaling Laws for Differentially Private Language Models](https://arxiv.org/abs/2501.18914)”, conducted in partnership with Google DeepMind, establishes laws that accurately model these intricacies, providing a complete picture of the compute-privacy-utility trade-offs. Guided by this research, we’re excited to introduce VaultGemma, the largest (1B-parameters), open model trained from scratch with differential privacy. We are releasing the weights on [Hugging Face](https://huggingface.co/google/vaultgemma-1b) and [Kaggle](https://www.kaggle.com/models/google/vaultgemma), alongside a [technical report](https://services.google.com/fh/files/blogs/vaultgemma_tech_report.pdf), to advance the development of the next generation of private AI.
+
+## Understanding the scaling laws
+
+With a carefully thought-out experimental methodology, we aimed to quantify the benefit of increasing model sizes, batch sizes, and iterations in the context of DP training. Our work required making some simplifying assumptions to overcome the exponential number of combinations one might consider trying. We assumed that how well the model learns depends mostly on the "noise-batch ratio” which compares the amount of random noise we add for privacy to the size of the data groups (batches) we use for training. This assumption works because the privacy noise we add is much greater than any natural randomness that comes from sampling the data.
+
+To establish a DP scaling law, we conducted a comprehensive set of experiments to evaluate performance across a variety of model sizes and noise-batch ratios. The resulting empirical data, together with known deterministic relationships between other variables, allows us to answer a variety of interesting scaling-laws–style queries, such as, “For a given compute budget, privacy budget, and data budget, what is the optimal training configuration to achieve the lowest possible training loss?”
```

Comment:
This is a hyper parameter search done once so we don't have to all do it again and again.